frontend/app/controllers/spree/order_contents_controller.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Spree
+  class OrderContentsController < Spree::StoreController
+    before_action :set_order, only: [:create]
+
+    def create
+      variant = Spree::Variant.find(params[:variant_id])
+      quantity = params[:quantity].present? ? params[:quantity].to_i : 1
+
+      if quantity_is_reasonable?(quantity)
```

Comment:
What about moving this logic (the `if`) into the `add_line_item_to_order` method? I feel like we don't have any benefit from having this check here and it's an easy opportunity to make the action code readability even leaner.

Also, I'm tempted to try to move this logic in the [`Spree::OrderContents`](https://github.com/solidusio/solidus/blob/2479b2466d217339d599fbc3a3490b7ddb1bd975/core/app/models/spree/order_contents.rb) class directly (with another PR). I mean, if we use that class in the rails console or programmatically to add stuff to an order we won't have that quantity check causing the error described in the code comment's issue? Any thoughts on this?

Reply:
Moving the `if` statement to `add_line_item_to_order` makes sense! I will do it👍 

I think moving the logic could be good also, for the reasons you stated and it would be nice to have this logic out of this controller. We will need any errors to go back up to the controller though, so will it be on `contents` instead or? 🤔 

If we do this in another PR, I think it could also be good to think about the future of the 'multiple `LineItem` functionality' we are going to have on `OrderContentsController#create`. Would it be more desirable that if one of the `LineItem`s do not have a reasonable quantity that we do not add any of them? Otherwise, it could be confusing for the User if only some of their chosen `LineItem`s get added. If so, I think we will need another method in [Spree::OrderContents](https://github.com/solidusio/solidus/blob/2479b2466d217339d599fbc3a3490b7ddb1bd975/core/app/models/spree/order_contents.rb)

Also, on that note of adding multiple variants and error messages 😅
1. Would it be an improvement if the error message said the `LineItem`'s name?
2. And if multiple `LineItem`s do not have a reasonable quantity, should it show one for each `LineItem` or one generic message for many?

Reply:
As you pointed out, [line items are currently saved when added](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/order_contents.rb#L98) so I think we would need a refactor to allow blocking the operation if any of the line item adding fails for some reason. 

For the error message, I think a more granular error messaging is preferable so I'd try to go with 2, let me know if it sounds doable.

Reply:
Thank you for your reply 😊 Would it be better if the `order.contents.add` could create multiple `Lineitem`s itself then? 

The error message sounds doable 👍

Reply:
> Would it be better if the `order.contents.add` could create multiple `Lineitems` itself then?

Yes, we could but the second parameter of `order.contents.add` is `quantity` and I think we can't assume all the items have to be of the same quantity when adding multiple products together.